gremlin-go/driver/result.go

```diff
@@ -21,18 +21,149 @@ package gremlingo
 
 // AN-968 Finish Result implementation.
 
-import "fmt"
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"strconv"
+)
 
 // Result Struct to abstract the Result and provide functions to use it.
 type Result struct {
 	result interface{}
 }
 
-// AsString returns the current Result formatted as a string.
+func newResult(result interface{}) *Result {
+	return &Result{result}
+}
+
+// AsString returns the current result item formatted as a string.
 func (r *Result) AsString() string {
 	return fmt.Sprintf("%v", r.result)
 }
 
-func newResult(result interface{}) *Result {
-	return &Result{result}
+// ToString returns the string representation of the Result struct in Go-syntax format
+func (r *Result) ToString() string {
+	return fmt.Sprintf("result{object=%v class=%T}", r.result, r.result)
+}
+
+// GetString gets the string representation of the result, equivalent to AsString()
+func (r *Result) GetString() string {
+	return r.AsString()
+}
+
+// GetInt gets the result by coercing it into an int, else returns an error if not parsable
+func (r *Result) GetInt() (int, error) {
+	return strconv.Atoi(r.AsString())
+}
+
+// GetByte gets the result by coercing it into a byte, else returns an error if not parsable
+func (r *Result) GetByte() (byte, error) {
+	res, err := strconv.ParseUint(r.AsString(), 10, 8)
+	if err != nil {
+		return 0, err
+	}
+	return byte(res), nil
+}
+
+// GetShort gets the result by coercing it into an int16, else returns an error if not parsable
+func (r *Result) GetShort() (int16, error) {
```

Comment:
Hmmm, we should probably rename these to be GetInt16, GetFloat32, etc. Otherwise Go programmers will be confused in the naming.

Reply:
Also perhaps we should add uint16/32/64 versions?

Reply:
Will do